shared-bindings/displayio/OnDiskGif.c

```diff
@@ -0,0 +1,210 @@
+//|       board.DISPLAY.show(splash)
+//|
+//|       odg = displayio.OnDiskBitmap('/sample.gif')
+//|       odg.play_frame() # Load the first frame
+//|       face = displayio.TileGrid(odg, pixel_shader=displayio.ColorConverter(input_colorspace=displayio.Colorspace.RGB565))
+//|       splash.append(face)
+//|       board.DISPLAY.refresh()
+//|
+//|       # Wait forever
+//|       while True:
+//|           gif.play_frame()
+//|           time.sleep(0.1)"""
+//|
+//|     def __init__(self, file: str) -> None:
+//|         """Create an OnDiskGif object with the given file.
+//|
+//|         :param file file: The name of the GIF file.
+//|
+//|         """
+//|         ...
+STATIC mp_obj_t displayio_ondiskgif_make_new(const mp_obj_type_t *type, size_t n_args, size_t n_kw, const mp_obj_t *all_args) {
+    mp_arg_check_num(n_args, n_kw, 1, 1, false);
+    mp_obj_t arg = all_args[0];
+
+    if (mp_obj_is_str(arg)) {
+        arg = mp_call_function_2(MP_OBJ_FROM_PTR(&mp_builtin_open_obj), arg, MP_ROM_QSTR(MP_QSTR_rb));
+    }
+
+    if (!mp_obj_is_type(arg, &mp_type_fileio)) {
+        mp_raise_TypeError(translate("file must be a file opened in byte mode"));
+    }
+
+    displayio_ondiskgif_t *self = m_new_obj(displayio_ondiskgif_t);
+    self->base.type = &displayio_ondiskgif_type;
+    common_hal_displayio_ondiskgif_construct(self, MP_OBJ_TO_PTR(arg));
+
+    return MP_OBJ_FROM_PTR(self);
+}
+
+//|     width: int
+//|     """Width of the gif. (read only)"""
+STATIC mp_obj_t displayio_ondiskgif_obj_get_width(mp_obj_t self_in) {
+    displayio_ondiskgif_t *self = MP_OBJ_TO_PTR(self_in);
+
+    return MP_OBJ_NEW_SMALL_INT(common_hal_displayio_ondiskgif_get_width(self));
+}
+
+MP_DEFINE_CONST_FUN_OBJ_1(displayio_ondiskgif_get_width_obj, displayio_ondiskgif_obj_get_width);
+
+MP_PROPERTY_GETTER(displayio_ondiskgif_width_obj,
+    (mp_obj_t)&displayio_ondiskgif_get_width_obj);
+
+//|     height: int
+//|     """Height of the gif. (read only)"""
+STATIC mp_obj_t displayio_ondiskgif_obj_get_height(mp_obj_t self_in) {
+    displayio_ondiskgif_t *self = MP_OBJ_TO_PTR(self_in);
+
+    return MP_OBJ_NEW_SMALL_INT(common_hal_displayio_ondiskgif_get_height(self));
+}
+
+MP_DEFINE_CONST_FUN_OBJ_1(displayio_ondiskgif_get_height_obj, displayio_ondiskgif_obj_get_height);
+
+MP_PROPERTY_GETTER(displayio_ondiskgif_height_obj,
+    (mp_obj_t)&displayio_ondiskgif_get_height_obj);
+
+//|     bitmap: Bitmap
+//|     """The bitmap used to hold the current frame."""
+STATIC mp_obj_t displayio_ondiskgif_obj_get_bitmap(mp_obj_t self_in) {
+    displayio_ondiskgif_t *self = MP_OBJ_TO_PTR(self_in);
+    return common_hal_displayio_ondiskgif_get_bitmap(self);
+}
+
+MP_DEFINE_CONST_FUN_OBJ_1(displayio_ondiskgif_get_bitmap_obj, displayio_ondiskgif_obj_get_bitmap);
+
+MP_PROPERTY_GETTER(displayio_ondiskgif_bitmap_obj,
+    (mp_obj_t)&displayio_ondiskgif_get_bitmap_obj);
+
+//|     play_frame: int
+//|     """Play next frame. Returns expected delay until the next frame."""
```

Comment:
What do you mean by play here?

Reply:
Updated this (and the other doc) to be more specific to what it does.